graylog2-web-interface/src/components/field-analyzers/FieldQuickValues.jsx

```diff
@@ -71,25 +80,44 @@ const FieldQuickValues = React.createClass({
   },
   _loadQuickValuesData() {
     if (this.state.field !== undefined) {
-      this.setState({ loadPending: true });
-      const promise = FieldQuickValuesStore.getQuickValues(this.state.field);
-      promise.then(data => this.setState({ data: data, loadPending: false }));
+      FieldQuickValuesActions.get(this.state.field, this.state.options.order, this.state.options.tableSize);
     }
   },
   _resetStatus() {
     this.setState(this.getInitialState());
   },
+
+  _onVizOptionsChange(newOptions) {
+    this.setState({ options: newOptions, showVizOptions: false }, () => this._loadQuickValuesData());
+  },
+
+  _showVizOptions() {
+    this.setState({ showVizOptions: true });
+  },
+
   render() {
     let content;
 
     let inner;
-    if (this.state.data.length === 0) {
+    if (this.state.showVizOptions) {
+      inner = (
+        <QuickValuesOptionsForm limit={this.state.options.limit}
```

Comment:
Adding this component as children of the `div` below, which sets `overflow: auto`, breaks the layout and makes this scroll horizontally.

Reply:
I will fix that.